src/glue/jobs/compare_parquet_datasets.py

```diff
@@ -0,0 +1,422 @@
+) -> list:
+    """This compares the row values between the two
+    datasets after the datasets have been reduced to the same dimensions
+    and outputs a message if any rows have differences"""
+    compare_msg = []
+    dataset_dict = keep_common_rows_cols(data_type, staging_dataset, main_dataset)
+    dataset_diff = dataset_dict["staging"].compare(
+        other=dataset_dict["main"], align_axis="columns", keep_equal=False
+    )
+    if not dataset_diff.empty:
+        compare_msg.append(
+            f"{data_type}: Staging dataset has value difference(s) with the main dataset."
+            f"Here is an example:\n{dataset_diff.head(1)}"
+        )
+    return compare_msg
+
+
+def get_data_types_to_compare(
+    s3: boto3.client, bucket_name: str, staging_namespace: str, main_namespace: str
+) -> list:
+    """This gets the common data types to run the comparison of the parquet datasets from
+    the two namespaced paths on based on the folders in the s3 bucket"""
+    staging_datatype_folders = get_folders_in_s3_bucket(
+        s3, bucket_name, namespace=staging_namespace
+    )
+    main_datatype_folders = get_folders_in_s3_bucket(
+        s3, bucket_name, namespace=main_namespace
+    )
+    return list(set(staging_datatype_folders + main_datatype_folders))
+
+
+def print_comparison_result(comparison_result: dict) -> None:
+    """"This prints the comparison result dictionary into a nice format"""
+    logger.info(logger.warning(f"Comparison results: {json.dumps(comparison_result)}"))
+    logger.info("Comparison results complete!")
+
+
+def compare_datasets_by_data_type(
+    args,
+    s3_filesystem: fs.S3FileSystem,
+    data_type: str,
+    comparison_result: dict,
+) -> dict:
+    """This runs the bulk of the comparison functions from beginning to end by data type"""
+    staging_dataset = get_parquet_dataset(
+        dataset_key=f"s3://{args.parquet_bucket}/{args.staging_namespace}/parquet/{data_type}/",
+        s3_filesystem=s3_filesystem,
+    )
+
+    main_dataset = get_parquet_dataset(
+        dataset_key=f"s3://{args.parquet_bucket}/{args.main_namespace}/parquet/{data_type}/",
+        s3_filesystem=s3_filesystem,
+    )
+    # check if one or both of the datasets have no data
+    if staging_dataset.empty or main_dataset.empty:
+        comparison_result[
+            data_type
+        ] = f"One of {args.staging_namespace} or {args.main_namespace} has no data. Comparison cannot continue."
+    else:
+        # check that the dataset has no dup cols or dup rows and that they have cols in common
+        comparison_result[data_type] = []
+        # check if one or both of the datasets have no data
+        if staging_dataset.empty or main_dataset.empty:
+            comparison_result["empty"][
+                data_type
+            ] = f"One of {args.staging_namespace} or {args.main_namespace} has no data. Comparison cannot continue."
+        else:
+            comparison_result[data_type].append(
```

Comment:
Still have to organize the structuring of the comparison messages within the dict. Trying to mirror/get inspiration from what we do in BridgeDownstream with the validate schema stuff.

Reply:
This will be mostly resolved with [datacompy](https://github.com/capitalone/datacompy)/follow its lead